core/src/libtx/proof.rs

```diff
@@ -57,6 +57,48 @@ where
 	))
 }
 
+/// Create a multisig bulletproof
+pub fn create_multisig<K, B>(
+	k: &K,
+	b: &B,
+	amount: u64,
+	key_id: &Identifier,
+	switch: SwitchCommitmentType,
+	common_nonce: &SecretKey,
+	tau_x: Option<&mut SecretKey>,
+	tau_one: Option<&mut PublicKey>,
+	tau_two: Option<&mut PublicKey>,
+	commits: &[Commitment],
+	step: u8,
+	extra_data: Option<Vec<u8>>,
+) -> Result<Option<RangeProof>, Error>
+where
+	K: Keychain,
+	B: ProofBuild,
+{
+	// TODO: proper support for different switch commitment schemes
```

Comment:
Doesn't the derive_key call on line 83 take care of the switch commitment? What more is needed?

Reply:
It does, and I copied this comment from the existing `create` function. I think it's referring to being able to specify the `SwitchCommitmentType` with a function argument.

I can add that if you want.